ingestion_server/ingestion_server/cleanup.py

```diff
@@ -50,6 +50,17 @@
 # may be inaccurate.
 TAG_MIN_CONFIDENCE = 0.90
 
+# We know that flickr and wikimedia support TLS, so we can add them here
+TLS_CACHE = {
+    "www.flickr.com": True,
+    "commons.wikimedia.org": True,
+    "https://www.eol.org/": True,
+    ".geograph.org.uk": True,
+    ".eol.org": True,
+    ".digitaltmuseum.org": True,
+    "www.geograph.org.uk": True,
+}
```

Comment:
How did the others get added? Are they similar to Flickr and Wikimedia in that we just know that they do support TLS?

If that's the case, would it be worth manually testing providers for this and adding them to the list (understanding how tedious that is)? Or, is it something we need to monitor/update over time due to the potential for this status to change (I suppose, most likely, that someone starts to support it that previously didn't)?

Would moving this into Redis make sense at all (as an entirely separate issue) or is there a different future change that would persist this TLS support status?

Reply:
I've added these manually, by looking through the logs and adding the ones that were being tested. Your logs suggest that `.musee-mccord.qc.ca` should have also been added :)

>If that's the case, would it be worth manually testing providers for this and adding them to the list (understanding how tedious that is)? Or, is it something we need to monitor/update over time due to the potential for this status to change (I suppose, most likely, that someone starts to support it that previously didn't)?

In short, I think that the TLS support check, the way it's done right now, should go away after we finish the cleanup step (1-2 refreshes to get the updated TSV and 1-2 update DAG runs could be enough).

We do not test the URLs that have insecure `http` scheme for TLS support.  The main reason we _were_ testing for TLS support was to add a best scheme to the URLs that _don't have it_ (to convert urls like `www.flickr.com/image/path` to **https://**`www.flickr.com/image/path`). There are not so many such rows in the database, mainly the ones that were ingested before the `ImageStore` improvements in the catalog, that were also not re-ingested. When we use the TSV from the cleanup run to update the catalog, all of the URLs will have a scheme, whether it is `http` or `https`, so the URL cleanup function will be parsing the URL and not running the TLS support checks because the scheme will not be `""`.

https://github.com/WordPress/openverse/blob/2646c5ead465603b42c70f58a190f7b50861d698/ingestion_server/ingestion_server/cleanup.py#L83-84

Do you think we should monitor for TLS support status? If so, I think this should be a separate issue. We could add a cleanup function to test domains with `http` for TLS support and report all of them, and then test them and update the URLs if the support changes.

Reply:
Thanks for the explanation and motivation for this feature.

> Do you think we should monitor for TLS support status?

If we're hand-maintaining the list then yes, I think we should revisit it periodically or else the cleanup step here will apply the incorrect transformations.

I agree that that is out of scope of this issue, _sort of_, but if we're expanding the list of sites we're automatically applying the transformation to, then it does make the matter slightly more pressing as the area of effect is slightly wider. The added providers are small though, I think, so it's negligible. In any case, I agree it's a separate issue. I wanted to mention it in case it needs to be explicitly documented as such in a GitHub issue.

If it's something that will go away soon though, due to some other mechanism that will render this step unnecessary, then we can ignore it altogether and just document in the code that it's a temporary hold-over.